src/configure/configurers/provisioningConfigurer.ts

```diff
@@ -0,0 +1,280 @@
+import { RemotePipelineTemplate } from '../model/templateModels';
+import { Messages } from '../resources/messages';
+import { TelemetryKeys } from '../resources/telemetryKeys';
+import { TracePoints } from '../resources/tracePoints';
+import { InputControl } from '../templateInputHelper/InputControl';
+import { IProvisioningConfigurer } from './IProvisioningConfigurer';
+
+// tslint:disable-next-line:interface-name
+interface DraftFile {
+    content: string;
+    path: string;
+    absPath: string;
+}
+
+const Layer: string = "ProvisioningConfigurer";
+export class ProvisioningConfigurer implements IProvisioningConfigurer {
+    private provisioningServiceClient: IProvisioningServiceClient;
+    private queuedPipelineUrl: string;
+    private refreshTime: number = 5 * 1000;
+    private maxNonStatusRetry: number = 60; // retries for max 5 min
+    private localGitRepoHelper: LocalGitRepoHelper;
+    private filesToCommit: DraftFile[] = [];
+
+    constructor(localGitRepoHelper: LocalGitRepoHelper){
+        this.localGitRepoHelper = localGitRepoHelper;
+    }
+
+    public async queueProvisioningPipelineJob(provisioningConfiguration: ProvisioningConfiguration, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration>{
+        try {
+            this.provisioningServiceClient =  await ProvisioningServiceClientFactory.getClient(wizardInputs.githubPATToken, wizardInputs.azureSession.credentials);
+            const OrgAndRepoDetails = wizardInputs.sourceRepository.repositoryId.split('/');
+            return await this.provisioningServiceClient.createProvisioningConfiguration(provisioningConfiguration, OrgAndRepoDetails[0], OrgAndRepoDetails[1]);
+        } catch (error){
+            telemetryHelper.logError(Layer, TracePoints.UnableToCreateProvisioningPipeline, error);
+            throw error;
+        }
+    }
+
+    public async getProvisioningPipeline(jobId: string, githubOrg: string, repository: string, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration>{
+       try {
+        this.provisioningServiceClient =  await ProvisioningServiceClientFactory.getClient(wizardInputs.githubPATToken, wizardInputs.azureSession.credentials);
+        return await this.provisioningServiceClient.getProvisioningConfiguration(jobId, githubOrg, repository);
+       } catch (error){
+            telemetryHelper.logError(Layer, TracePoints.UnabletoGetProvisioningPipeline, error);
+            throw error;
+       }
+    }
+
+    public async awaitProvisioningPipelineJob(jobId: string, githubOrg: string, repository: string, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration> {
+        let statusNotFound: number = 0;
+        try {
+            const provisioningServiceResponse = await this.getProvisioningPipeline(jobId, githubOrg, repository, wizardInputs);
+            if ( provisioningServiceResponse.result){
```

Comment:
do we need to check for `if ( provisioningServiceResponse && provisioningServiceResponse.result)`

Reply:
added this